v2.0.0

saket opened the pull request:
### Improvements
- v1 made an assumption that `InboxRecyclerView` and its `ExpandablePageLayout` will always share the same parent and have the same width & height. v2 removes this restriction. (Fixes #10).
- It'll now be possible to detach a `ExpandablePageLayouts` from its `InboxRecyclerView`, essentially making it possible to use multiple expandable pages with the same list. (Fixes #32).

### Breaking changes
- `InboxRecyclerView.setExpandablePage()` is removed in favor of using `InboxRecyclerView.expandablePage` directly. Kotlin FTW! If you wish to specify the page's collapse threshold then you can again set `ExpandablePageLayout.pullToCollapseThresholdDistance` directly.
- ~`PageStateChangeCallbacks` functions were updated to include a `page: ExpandablePageLayout` parameter. I'm not too happy with this change so I'll try reverting this.~

saket committed:
Add support for detaching ExpandablePageLayout from InboxRecyclerView

saket committed:
Leverage Kotlin property setters for attaching and detaching pages

saket committed:
Kill InboxRecyclerView#setExpandablePageLayout() in favor of direct property access

saket committed:
Rename InboxRecyclerView#page -> expandablePage

saket committed:
Correctly do reverse for loops in Kotlin

saket committed:
Handle item expansions when the list and its page have different bounds and parents

saket committed:
Rename ExpandedItem#expandedItemLocationRect -> #locationOnScreen

saket committed:
Rename ExpandedItem#itemId -> #adapterId

saket committed:
Reformat some code

saket committed:
Rename @dimen/pull_collapsible_activity_elevation -> irv_pull_collapsible_activity_elevation

ZacSweers commented on inboxrecyclerview/src/main/java/me/saket/inboxrecyclerview/page/ExpandablePageLayout.kt:
```diff
@@ -489,8 +538,9 @@ open class ExpandablePageLayout @JvmOverloads constructor(
-    for (i in stateChangeCallbacks.indices.reversed()) {
-      stateChangeCallbacks[i].onPageAboutToExpand(expandAnimDuration)
+    // Reverse loop to let listeners remove themselves while in the loop.
+    for (i in stateChangeCallbacks.size - 1 downTo 0) {
```
This would have the same issue as before since it's still just indexing based off the original size. You either need to back it with a `CopyOnWriteArrayList` + for-each, or make a defensive copy.

saket replied:
Done: https://github.com/saket/InboxRecyclerView/pull/34/commits/6c5c7b7f81b11fe13ec0c8cea02ad1a1371a0487.

saket committed:
Use CopyOnWriteArrayList for storing page callbacks

Reverse-for loops avoided object allocation while allowing callbacks to remove
themselves, but weren't foolproof. CopyOnWriteArrayList returns a new copy of
the underlying array on every iteration so it's safer. It's also more efficient
than using Kotlin's immutable list because it avoids blanket creating a new list
with extra capacity everytime a callback is added or removed.

saket committed:
Refactor UncoveredAreaTintPainter to avoid using nullable fields

saket committed:
Remove unused VERSION_CODE
